runtime/recipe/type-checker.js

```diff
@@ -11,21 +11,149 @@ import assert from '../../platform/assert-web.js';
 
 class TypeChecker {
 
-  // list: [{type, direction, connection}]
-  static processTypeList(list) {
-    if (list.length == 0) {
-      return {type: {type: undefined}, valid: true};
+  // resolve a list of handleConnection types against a handle
+  // base type.
+  static processTypeList(baseType, list) {
```

Comment:
Brief documentation on this and other methods below? I think it would help folks discover/know when to use these, when they're engaged in type related logic elsewhere and poking about looking for helpers.

Reply:
Done. However, a word of caution is: don't use these. They have a pretty specific purpose and should only be during type resolution, which happens at a defined time.